### PR TITLE
STELLAR-757 Use Cobra to allow passing in of parameters

### PR DESCRIPTION
**Pull request by reiinakano**
Runs GRPC server with `starcoder serve -bindAddress <> -flowgraphDIr <>`

If no flowgraphDir is given, default to packed in flowgraphs.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	} else {
+		// Find home directory.
+		home, err := homedir.Dir()
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		// Search config in home directory with name ".starcoder" (without extension).
+		viper.AddConfigPath(home)
```
Comment (anuraaga) [inline cmd/root.go]: How about `AddConfigPath(home/.starcoder)` and `SetConfigName(config)`?

File: cmd/serve.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+// serveCmd represents the serve command
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Start the Starcoder server",
+	Long: `Start the Starcoder server`,
+	Run: func(cmd *cobra.Command, args []string) {
+		log.Printf("serve called, using bind address %v", serveCmdConfig.BindAddress)
+
+		if serveCmdConfig.FlowgraphDir == "" {
+			tempDir, err := ioutil.TempDir("", "starcoder")
```
Comment (anuraaga) [inline cmd/serve.go]: `defer delete the temp dir`?

File: build.gradle [hunk 3]
```diff
@@ -125,6 +125,21 @@ task depGet(type: com.github.blindpirate.gogradle.Go) {
...
+task getRice(type: com.github.blindpirate.gogradle.Go) {
+    go 'get -u github.com/GeertJohan/go.rice/...'
+}
+
+task useRice(type:Exec) {
+    inputs.dir 'flowgraphs'
+    outputs.file 'cmd/rice-box.go'
+
+    workingDir project.file('cmd').toString()
+    dependsOn getRice
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply build.gradle]: First means first (the dependency graph is the most important part of a build so we put it as the first line of a task, see the other ones)

File: cmd/serve.go [hunk 4]
```diff
@@ -74,9 +78,33 @@ var serveCmd = &cobra.Command{
...
+		go func() {
+			for {
+				select {
+				case s := <-sigs:
+					switch s {
+					case syscall.SIGINT:
+						fallthrough
+					case syscall.SIGTERM:
+						fallthrough
+					case syscall.SIGQUIT:
```
Comment (anuraaga) [inline cmd/serve.go]: Can you use `defer` instead?

Comment (reiinakano) [reply cmd/serve.go]: Doesn't work, I don't think it gets called with SIGINT.

Comment (reiinakano) [reply cmd/serve.go]: Yup, that worked.

**Pull request merged**